### PR TITLE
Move `DoFHandler::prepare_c_and_r()` to `hp::Refinement::limit_p_level_difference()`.

### PR DESCRIPTION
**Pull request by marcfehling**
Since `DoFHandler::prepare_c_and_r()` does not need access to private member functions, and only makes sense in the hp-adaptation context, this PR will move the former member function to become the now free function `hp::Refinement::limit_plevel_difference()`.

### CODE REVIEW

File: doc/news/changes/minor/20210223Fehling [hunk 1]
```diff
@@ -1,4 +1,4 @@
-New: Member function DoFHandler::prepare_coarsening_and_refinement()
+New: Function hp::Refinement::limit_plevel_difference()
```
Comment (peterrum) [inline doc/news/changes/minor/20210223Fehling:1]:
```
New: Function hp::Refinement::limit_p_level_difference()
```
?

Are the other functions with similar names (h/p) in the library?

**Review by peterrum — APPROVED** [review]

File: include/deal.II/hp/refinement.h [hunk 2]
```diff
@@ -647,6 +648,47 @@ namespace hp
     /**
      * @}
      */
+
+    /**
+     * @name Optimiize p-level distribution
```
Comment (bangerth) [inline include/deal.II/hp/refinement.h:653]:
```suggestion
     * @name Optimize p-level distribution
```

File: include/deal.II/hp/refinement.h [hunk 3]
```diff
@@ -647,6 +648,47 @@ namespace hp
...
+     * The function can optionally be called before performing adaptation with
+     * Triangulation::execute_coarsening_and_refinement(). It is not necessary
+     * to call this function, nor will it be automatically invoked in any part
+     * of the library (contrary to its Triangulation counterpart).
+     *
+     * Returns whether any future FE indices have been changed by this function.
+     */
+    template <int dim, int spacedim>
+    bool
+    limit_plevel_difference(
```
Comment (bangerth) [inline include/deal.II/hp/refinement.h:684]: Like @peterrum I would suggest `limit_p_level_difference()` for the name.

**Review by bangerth — APPROVED** [review]

File: doc/news/changes/minor/20210223Fehling (hunk 1 above)
Comment (marcfehling) [reply doc/news/changes/minor/20210223Fehling:1]: There are a few in the `hp::Refinement` namespace, like `choose_p_over_h`.

`prepare_c_and_r` was simply wrong in this context as it is not a mandatory preparation. The function should rather describe what it's doing.

**Comment by marcfehling** [comment]
After changing the function name as requested, I'll go on and merge this PR to prepare a follow-up, as this PR really only renames the function, we have two positive reviews, and the testsuite passes.

**Pull request merged**